### PR TITLE
blockdev_stream_with_ioerror:stream after block error

### PR DESCRIPTION
**Pull request by aliang123**
Do block stream after block error(read/write/no space)

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1994868

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_stream_with_ioerror.q35: PASS (103.50 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks.

File: qemu/tests/blockdev_stream_with_ioerror.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        """Generate temp data file in guest"""
+        self.session = self.main_vm.wait_for_login()
+        file_path = "%s/%s" % (root_dir, filename)
+        count = int(
+            utils_numeric.normalize_data_size(
+                size,
+                order_magnitude="M",
+                factor=1024))
+        dd_cmd = self.main_vm.params.get(
+            "dd_cmd", "dd if=/dev/urandom of=%s bs=1M count=%s oflag=direct")
```
Comment (zhencliu) [inline qemu/tests/blockdev_stream_with_ioerror.py:22]: same comment from #2973

File: qemu/tests/blockdev_stream_with_ioerror.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        mk_file_cmd = dd_cmd % (file_path, count)
+        try:
+            self.session.cmd(mk_file_cmd, timeout=timeout)
+        except ShellTimeoutError as e:
+            if self.params["error_msg"] not in str(e):
+                self.test.fail(str(e))
```
Comment (zhencliu) [inline qemu/tests/blockdev_stream_with_ioerror.py:28]: same comment from #2973

File: qemu/tests/cfg/blockdev_stream_with_ioerror.cfg [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    start_vm = yes
+    storage_pools = default
+    storage_type_default = "directory"
+    storage_pool = default
+    base_tag = "data"
+    image_size_data = 2G
+    snapshot_tag = sn1
+    image_size_sn1 = 2G
+    image_name_sn1 = sn1
+    image_format_sn1 = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_stream_with_ioerror.cfg:16]: set image_format_data = qcow2 for only qcow2 can be created in the space whose size is less than the qcow2 image

File: qemu/tests/cfg/blockdev_stream_with_ioerror.cfg [hunk 4]
```diff
@@ -0,0 +1,52 @@
+- blockdev_stream_with_ioerror:
+    type = blockdev_stream_with_ioerror
+    virt_test_type = qemu
+    images += " data"
+    force_create_image_data = yes
+    force_remove_image_data = yes
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_stream_with_ioerror.cfg:6]: remove_image_data = yes

File: qemu/tests/blockdev_stream_with_ioerror.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        if not self.session:
+            self.session = self.main_vm.wait_for_login()
+        output = self.session.cmd_output('\n', timeout=120)
+        if self.params["dd_done"] not in output:
+            self.test.fail("dd not continue to run after vm resume")
+        for info in self.files_info:
+            file_path = "%s/%s" % (info[0], info[1])
+            md5_cmd = "md5sum %s > %s.md5 && sync" % (file_path, file_path)
+            self.session.cmd(md5_cmd, timeout=120)
+        self.session.close()
```
Comment (zhencliu) [inline qemu/tests/blockdev_stream_with_ioerror.py:49]: same comments as #2973

File: qemu/tests/cfg/blockdev_stream_with_ioerror.cfg [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    start_vm = yes
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_stream_with_ioerror.cfg:7]: It's better to add 'kill_vm = yes'

File: qemu/tests/cfg/blockdev_stream_with_ioerror.cfg [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    pre_command += "pvcreate `cat ${loopdev}` &&"
+    pre_command += "vgcreate ${vgname} `cat ${loopdev}` &&"
+    pre_command += "lvcreate -L 100M -n ${lvname} ${vgname} &&"
+    pre_command += "mkfs -t ext3 /dev/${vgname}/${lvname} && mkdir -p ${mount_dir} &&"
+    pre_command += "mount /dev/${vgname}/${lvname} ${mount_dir} &&"
+    pre_command += "qemu-img create -f qcow2 ${image_name_data} ${image_size_data}"
+
+    post_command += "umount ${mount_dir} &&"
+    post_command += "lvremove -f /dev/${vgname}/${lvname} &&"
+    post_command += "vgremove -f ${vgname} &&"
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_stream_with_ioerror.cfg:45]: Same comment as #2976

File: qemu/tests/cfg/blockdev_stream_with_ioerror.cfg [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    !Host_RHEL.m7:
+        node = "drive_data"
+        overlay = "drive_sn1"
+        qemu_force_use_drive_expression = no
+
+    tmpdir = /home/tmp
+    mount_dir = /mnt/test
+    loopdev = ${tmpdir}/loopdev
+    backend_base = ${tmpdir}/disk.img
+    image_name_data = ${mount_dir}/data.qcow2
```
Comment (ngu-niny) [inline qemu/tests/cfg/blockdev_stream_with_ioerror.cfg:30]: The image name in test log is '/mnt/test/data.qcow2.qcow2'.

**Comment by ngu-niny** [comment]
1-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_stream_with_ioerror Finshed 2021-08-30 00:24:39 2021-08-30 00:26:32 PASS 112.612344

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**